sstables_loader.cc

```diff
@@ -126,23 +126,29 @@ using unlink_sstables = bool_class<struct unlink_sstables_tag>;
 
 class sstable_streamer {
 protected:
+    using stream_scope = sstables_loader::stream_scope;
     netw::messaging_service& _ms;
     replica::database& _db;
     replica::table& _table;
     locator::effective_replication_map_ptr _erm;
     std::vector<sstables::shared_sstable> _sstables;
     const primary_replica_only _primary_replica_only;
     const unlink_sstables _unlink_sstables;
+    const stream_scope _stream_scope;
 public:
-    sstable_streamer(netw::messaging_service& ms, replica::database& db, ::table_id table_id, std::vector<sstables::shared_sstable> sstables, primary_replica_only primary, unlink_sstables unlink)
+    sstable_streamer(netw::messaging_service& ms, replica::database& db, ::table_id table_id, std::vector<sstables::shared_sstable> sstables, primary_replica_only primary, unlink_sstables unlink, stream_scope scope)
             : _ms(ms)
             , _db(db)
             , _table(db.find_column_family(table_id))
             , _erm(_table.get_effective_replication_map())
             , _sstables(std::move(sstables))
             , _primary_replica_only(primary)
             , _unlink_sstables(unlink)
+            , _stream_scope(scope)
     {
+        if (_primary_replica_only == primary_replica_only::yes && _stream_scope != stream_scope::all) {
```

Comment:
nit: as a bool_class `if (_primary_replica_only && ...)` would be enough, as it has a bool operator.

Reply:
LGTM otherwise

Reply:
Yes, nice point, applied